react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class NewsletterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { email: '' };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ email: event.target.value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    console.log(this.state);
+  }
+
+  render() {
+    return (
+    <form action="">
+        Email
+        <input type="text" onChange={this.handleChange} value={this.state.email} />
+        <button onClick={this.handleSubmit}>Sign Up</button>
```

Comment:
In forms, we listen to submit events on the `<form>` instead of click events on the `<button>`.